firefox-ios/Storage/Rust/RustAutofill.swift

```diff
@@ -475,17 +486,73 @@ public class RustAutofill {
         }
     }
 
-    private func handleKeyCorruption() {
-        logger.log("Autofill key was corrupted, new one generated",
-                   level: .warning,
-                   category: .storage)
-        scrubCreditCardNums(completion: { _, _ in })
+    private func getKeychainData(rustKeys: RustAutofillEncryptionKeys,
+                                 completion: @escaping (String?, String?) -> Void) {
+        DispatchQueue.global(qos: .background).sync {
```

Comment:
Is there a specific reason this needs to be `sync` (vs `async` and allowing the function to return after dispatching)?

Reply:
When I created [similar changes for rust logins](https://github.com/mozilla-mobile/firefox-ios/pull/18401/files#diff-862d697d81bbc65c116f7426f3c68f0b0881f0aa0b09919c2ac1cecfcd0c4c46) I initially made these key calls in async dispatch queues. That caused a number of unit tests to fail, so I switched the queues to sync.

It might be that the logic in this function and/or the `getStoredKey` function, where this function is called, could be refactored so an async queue can be used here without any regressions. If you have suggestions, I'm open.